compiler/rustc_parse/src/lexer/mod.rs

```diff
@@ -268,6 +268,9 @@ impl<'a> StringReader<'a> {
                 // tokens like `<<` from `rustc_lexer`, and then add fancier error recovery to it,
                 // as there will be less overall work to do this way.
                 let token = unicode_chars::check_for_substitution(self, start, c, &mut err);
+                if c == '\x00' {
+                    err.help("source files must be encoded in UTF-8, unexpected null bytes might occur when the wrong text encoding is used");
```

Comment:
I was confused when I saw "wrong", maybe do we mean a "different" text encoding?